File: gex/src/model/BarUserSkillChanges.ts

export class BarUserSkillChanges {
    public userID: number = 0;
    public gamemodes: BarUserSkillGamemode[] = [];

    public static parse(elem: any): BarUserSkillChanges {
        return {
            userID: elem.userID,
            gamemodes: elem.gamemodes.map((iter: any) => BarUserSkillGamemode.parse(iter))
        };
    }
}

export class BarUserSkillGamemode {
    public gamemode: number = 0;
    public changes: BarUserSkillChangeEntry[] = [];

    public static parse(elem: any): BarUserSkillGamemode {
        return {
            gamemode: elem.gamemode,
            changes: elem.changes.map((iter: any) => BarUserSkillChangeEntry.parse(iter))
        };
    }
}

export class BarUserSkillChangeEntry {
    public skill: number = 0;
    public skillUncertainty: number = 0;
    public timestamp: Date = new Date();

    public static parse(elem: any): BarUserSkillChangeEntry {
        return {
            skill: elem.skill,
            skillUncertainty: elem.skillUncertainty,
            timestamp: new Date(elem.timestamp)
        };
    }
}

File: gex/Models/UserStats/BarUserSkillChanges.cs
using System;
using System.Collections.Generic;

namespace gex.Models.UserStats {

    /// <summary>
    ///     represents all changes of a user
    /// </summary>
    public class BarUserSkillChanges {

        /// <summary>
        ///     ID of the user these changes are for
        /// </summary>
        public long UserID { get; set; }

        /// <summary>
        ///     list of all gamemode skill changes
        /// </summary>
        public List<BarUserSkillGamemode> Gamemodes { get; set; } = [];

    }

    /// <summary>
    ///     represents the changes in skill for a single gamemode
    /// </summary>
    public class BarUserSkillGamemode {

        /// <summary>
        ///     ID of the gamemode
        /// </summary>
        public byte Gamemode { get; set; }

        /// <summary>
        ///     all changes to skill for this gamemode
        /// </summary>
        public List<BarUserSkillChangeEntry> Changes { get; set; } = [];

    }

    /// <summary>
    ///     represents a single change in a players skill for a specific gamemode
    /// </summary>
    public class BarUserSkillChangeEntry {

        /// <summary>
        ///     skill value
        /// </summary>
        public double Skill { get; set; }

        /// <summary>
        ///     mau
        /// </summary>
        public double SkillUncertainty { get; set; }

        /// <summary>
        ///     when this change took place
        /// </summary>
        public DateTime Timestamp { get; set; }

    }

}
